### PR TITLE
Remove DefaultForTaskType field from plugin entries

### PR DESCRIPTION
**Pull request by hamersaw**
# TL;DR
The DefaultForTaskType field is not necessary for a more simple task registration scheme (described in more depth in the linked issue). In every plugin in this repository it is identical to the RegisteredTaskTypes field.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [x] Any pending items have an associated Issue

## Complete description
^^^

## Tracking Issue
https://github.com/flyteorg/flyte/issues/2205

## Follow-up issue
_NA_

### CODE REVIEW

File: go/tasks/config/config.go [hunk 1]
```diff
@@ -18,12 +16,7 @@ var (
 
 // Top level plugins config.
 type Config struct {
-	EnabledPlugins []string `json:"enabled-plugins" pflag:"[]string{\"*\"},List of enabled plugins, default value is to enable all plugins."`
-}
-
-func (cfg Config) IsEnabled(pluginToCheck string) bool {
-	return cfg.EnabledPlugins != nil && len(cfg.EnabledPlugins) >= 1 &&
-		(cfg.EnabledPlugins[0] == "*" || utils.Contains(cfg.EnabledPlugins, pluginToCheck))
+	EnabledPlugins []string `json:"enabled-plugins" pflag:"[]string{\"*\"},Deprecated - List of enabled plugins, default value is to enable all plugins."`
```
Comment (EngHabu) [inline go/tasks/config/config.go:19]:
```suggestion
	EnabledPlugins []string `json:"enabled-plugins" pflag:",Deprecated - List of enabled plugins, default value is to enable all plugins."`
```

Comment (EngHabu) [reply go/tasks/config/config.go:19]: The style of setting default values inline is deprecated (haven't been fully removed from all the codebase yet admittedly).

Instead:
```golang
//go:generate pflags Config --default-var=defaultConfig

var (
  defaultConfig = &Config{
    EnabledPlugins: []string{},
  }

  rootSection = config.MustRegisterSection(configSectionKey, &defaultConfig)
)

File: go/tasks/config/config.go [hunk 2]
```diff
@@ -2,28 +2,25 @@ package config
...
 
 // Top level plugins config.
 type Config struct {
-	EnabledPlugins []string `json:"enabled-plugins" pflag:"[]string{\"*\"},List of enabled plugins, default value is to enable all plugins."`
-}
-
-func (cfg Config) IsEnabled(pluginToCheck string) bool {
-	return cfg.EnabledPlugins != nil && len(cfg.EnabledPlugins) >= 1 &&
-		(cfg.EnabledPlugins[0] == "*" || utils.Contains(cfg.EnabledPlugins, pluginToCheck))
+	EnabledPlugins []string `json:"enabled-plugins" pflag:",Deprecated - List of enabled plugins, default value is to enable all plugins."`
```
Comment (EngHabu) [inline go/tasks/config/config.go:23]: Should we delete this field altogether?

**Review by EngHabu — DISMISSED** [review]

File: go/tasks/config/config.go (hunk 2 above)
Comment (hamersaw) [reply go/tasks/config/config.go]: I think that makes sense. Done.

**Review by EngHabu — APPROVED** [review]

**Pull request merged**